src/app/modules/lti/pages/lti/lti.component.ts

```diff
@@ -27,41 +35,76 @@ const noChildError = new Error(LTIError.NoChild);
 })
 export class LTIComponent implements OnDestroy {
 
+  private userId$ = this.activatedRoute.queryParamMap.pipe(
+    map(queryParams => queryParams.get(userIdParam)),
+  );
+
+  private isRedirection$ = this.activatedRoute.queryParamMap.pipe(
+    map(queryParams => queryParams.has(isRedirectionParam)),
+  );
+
   private contentId$ = this.activatedRoute.paramMap.pipe(
     map(params => {
       const contentId = params.get('contentId');
       if (!contentId) throw new Error('unexpected: contentId should be defined');
       return contentId;
     }),
   );
-  readonly navigationData$ = this.contentId$.pipe(
-    switchMap(contentId => this.getNavigationData(contentId).pipe(mapToFetchState())),
+
+  private isLoggedIn$ = combineLatest([ this.userSession.userProfile$, this.userId$ ]).pipe(
+    map(([ profile, userId ]) => !!userId && profile.groupId === userId),
+    take(1),
```

Comment:
So if it is only to keep the first value, is it relevant to use Rx `userId$` and `isRedirection$` for that ? Could you just use the activated route snapshot to compute it once?
Or you assume parameters may change (they may probably)... but so the `take(1)` makes the component inconsistent in this case.

Reply:
I removed the `take` operator
fixed